From c5e08730103eaaa44f11cda9294938ca1dc762f9 Mon Sep 17 00:00:00 2001
From: Lionel Debieve <lionel.debieve@foss.st.com>
Date: Tue, 22 Jun 2021 00:58:05 +0200
Subject: [PATCH 1108/1695] libutee: tui: force image alpha color to 0xFF

Put the alpha mask to FF when CFG_STM32_LTDC is selected.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git ec8a984badfba605d46c0f93355b9620a78bedea libutee: tui: force image alpha color to 0xFF]
Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Change-Id: Ic0c303c920d3f157b994e7508df628018b045fb3
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/392609
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 lib/libutee/tui/image.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/libutee/tui/image.c b/lib/libutee/tui/image.c
index 6b884673d..29295d7e6 100644
--- a/lib/libutee/tui/image.c
+++ b/lib/libutee/tui/image.c
@@ -9,8 +9,8 @@
 
 static uint32_t color_to_pixel(uint32_t color)
 {
-	/* Convert from ARGB to RGBA and byte swap */
-	return TEE_U32_BSWAP((color >> 24) | (color << 8));
+	/* Force image alpha color */
+	return color | 0xFF000000;
 }
 
 struct image *image_alloc(size_t width, size_t height, uint32_t color)
-- 
2.47.2

